### PR TITLE
FIX: incremental  estimators tests

### PR DESCRIPTION
**Pull request by samir-nasibli**
### Description

fixes incremental estimators tests.
Output array should be converted to numpy array before setting tol dependent on the dtype, otherwise the tol settings will not be set. Found on #1861 

---

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.  
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [ ] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

**Review by icfaust — COMMENTED** [review]
I tried to suggest this when it was first implemented, but was overridden.  Rather than cast it to numpy for checking dtype, couldn't we just use the dtype test parameter? i.e.
``` tol = 2e-6 if dtype == np.float32 else 1e-7```?

**Comment by samir-nasibli** [comment]
> Rather than cast it to numpy for checking dtype, couldn't we just use the dtype test parameter? i.e. ` tol = 2e-6 if dtype == np.float32 else 1e-7`?

All asserts require numpy inputs, so make sense convert them to numpy ndarrays.

File: sklearnex/preview/decomposition/tests/test_incremental_pca.py [hunk 1]
```diff
@@ -200,7 +200,7 @@ def test_sklearnex_partial_fit_on_gold_data(dataframe, queue, whiten, num_blocks
 
     X_df = _convert_to_dataframe(X, sycl_queue=queue, target_df=dataframe)
     transformed_data = incpca.transform(X_df)
-    check_pca_on_gold_data(incpca, dtype, whiten, transformed_data)
+    check_pca_on_gold_data(incpca, dtype, whiten, _as_numpy(transformed_data))
```
Comment (olegkkruglov) [inline sklearnex/preview/decomposition/tests/test_incremental_pca.py:203]: similar comment here, it's better to use dtype arg from check_pca_on_gold_data in the function for tolerance setting, it allows to avoid additional _as_numpy call

Comment (olegkkruglov) [reply sklearnex/preview/decomposition/tests/test_incremental_pca.py:203]: _as_numpy call already exists inside the method, only tolerance setting is necessary to be updated using dtype arg

**Comment by olegkkruglov** [comment]
> Rather than cast it to numpy for checking dtype, couldn't we just use the dtype test parameter? i.e. ` tol = 2e-6 if dtype == np.float32 else 1e-7`?

agree, this is exactly what I asked in my comments

**Review by olegkkruglov — APPROVED** [review]
LGTM, assuming green CI

**Comment by samir-nasibli** [comment]
Last CI run: http://intel-ci.intel.com/ef691edb-824e-f15b-993d-a4bf010d0e2e

**Comment by samir-nasibli** [comment]
@mergify backport rls/2024.7.0-rls
